src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -58,27 +57,18 @@ export class HttpResourceFetcher implements ResourceFetcher {
         return this.params.showUrl ? this.url : this.params.friendlyName ?? 'resource from URL'
     }
 
-    private async getResponseFromGetRequest(): Promise<request.Response> {
-        return new Promise<request.Response>((resolve, reject) => {
-            const call = request(
-                {
-                    url: this.url,
-                    headers: { 'User-Agent': VSCODE_EXTENSION_ID.awstoolkit },
-                },
-                (err, response, body) => {
-                    if (err) {
-                        // swallow error to keep URL private
-                        // some AWS APIs use presigned links (e.g. Lambda.getFunction); showing these represent a securty concern.
-                        reject({ code: err.code })
-                    }
-                    resolve(response)
-                }
-            )
-
-            if (this.params.pipeLocation) {
-                call.pipe(fs.createWriteStream(this.params.pipeLocation))
-            }
+    private async getResponseFromGetRequest(): Promise<Response<string>> {
+        const response = await got(this.url, {
```

Comment:
The pipe now happens sequentially rather than in parallel. Using `got` is a tad bit more cumbersome when you want to pipe the results somewhere as well as return them. I think it'd be wise to just return the pipe itself in the `pipeLocation` case, that way callers can wait for the stream to finish.